framework/helpers/channels_test.go

```diff
@@ -0,0 +1,78 @@
+package helpers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/launchdarkly/sdk-test-harness/framework/opt"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNonBlockingSend(t *testing.T) {
+	ch1 := make(chan string)
+	assert.False(t, NonBlockingSend(ch1, "a"))
+
+	ch2 := make(chan string, 1)
+	assert.True(t, NonBlockingSend(ch2, "a"))
+	assert.Equal(t, "a", <-ch2)
+}
+
+func TestTryReceive(t *testing.T) {
+	ch := make(chan string, 1)
+	assert.Equal(t, opt.None[string](), TryReceive(ch, time.Millisecond))
```

Comment:
Can the compiler infer the generic parameter of string here in `opt.None`? (genuine question - I don't know if this is a feature or not.)

Reply:
It wouldn't be able to infer it in this particular line no matter what, because the parameter type of `assert.Equal` is just `interface{}` (i.e. there's no compile-time checking that the 2nd & 3rd parameters have the same type - a frequent pitfall in tests that use frameworks like that, and my `matchers` stuff has the same problem, unless & until I rewrite it with generics).

Anyway, the general point was a good question and I didn't know the answer, so I tried and the answer is alas no:

```go

func foo(m opt.Maybe[string]) {}

func bar() {
	foo(opt.None())
	    ^^ compiler error: "cannot infer V"
}
```